bl31/bl31.mk

```diff
@@ -24,6 +24,7 @@ BL31_SOURCES		+=	bl31/bl31_main.c				\
 				bl31/bl31_context_mgmt.c			\
 				common/runtime_svc.c				\
 				lib/aarch64/setjmp.S				\
+				lib/cpus/aarch64/dsu_helpers.S			\
```

Comment:
CPU libs are usually pulled in by the platforms that need them, not generic code. Also, what about the BL2_AT_EL3 case?

Reply:
1) Firstly, the DSU erratum does not apply only to a specific CPU. Secondly, the workaround is the same for every CPU it applies to. For these reasons it was implemented in a separate file, as a helper function.
Subsequently, the "dsu_helpers.S" file is included by generic code, similar to how we include "cpu_helpers.S".
However, the strategy of when to *apply* the workaround is exactly the same as with other workarounds and indeed is controlled by the platform.mk files. Conditional branches inside the relevant cortex_aXX.S files have been added so as to control when the workaround code will be executed.

2) For the BL2_AT_EL3 case, you are right, will be addressed.

Reply:
@jts-arm, OK